.github/workflows/rust.yml

```diff
@@ -3,28 +3,58 @@ name: Rust
 on: [push]
 
 jobs:
-  build:
+
+  codestyle:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Rust
+        uses: hecrj/setup-rust-action@v1
+        with:
+          components: rustfmt
+          rust-version: nightly
+      - uses: actions/checkout@v2
+      - run: cargo fmt -- --check --config-path <(echo 'license_template_path = "HEADER"')
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Rust
+        uses: hecrj/setup-rust-action@v1
+        with:
+          components: clippy
+      - uses: actions/checkout@v2
+      - run: cargo clippy --all-targets -- -D clippy::all
+
+  compile:
     runs-on: ubuntu-latest
+    steps:
+      - name: Set up Rust
+        uses: hecrj/setup-rust-action@v1
+      - uses: actions/checkout@master
+      - run: cargo check --all
```

Comment:
What's the purpose of having a separate `compile` build? Why are we running with `--all`? (I don't know what it means and the help says "Alias for --workspace (deprecated); Check all packages in the workspace")

Reply:
I don't think this is required, but I do see it commonly. I think the idea is that you can fail the test suite faster this way when encountering certain kinds of errors.

`--all` is probably not required here.

Reply:
I see. In practice, the build errors on CI are really rare (in fact, I don't remember that ever happening in this project) so it seems like it just slows down the build by 30 seconds.

Reply:
Ah, is this the slowest job? Otherwise it should be running in parallel and not actually contribute to the overall workflow runtime.

Reply:
As I understand, the current workflow is something like this:
```
 /---- codestyle [15s] ---\       /----- test + coverage [6m] ----\
* ---- lint [40s] --------*------*------ test (beta)  [1m] -------*
 \---- compile [45s] -----/       \-------- test (nightly) [1m] --/
```

As long as any of the jobs in the first block prevent the second block from running, the overall run time is slower. Of those I can see why we might need `rustfmt` and `lint`, but `cargo check` seems like a waste of energy.

Reply:
I decoupled the job dependencies so they'll now all run in parallel with one another.